### PR TITLE
add onclick listener for purchase

### PR DESCRIPTION
**Pull request by sabrina-li**
handle on click for purchase button
validate - legal input for the customer info; subtotal >0

for demo purpose only:
if success -toast success
if fail - toast fail

### CODE REVIEW

File: java/app/src/main/java/com/fullstorydev/shoppedemo/ui/checkout/CheckoutFragment.java [hunk 1]
```diff
@@ -60,8 +62,25 @@ public void onActivityCreated(@Nullable Bundle savedInstanceState) {
...
+                binding.setHandlers(this);
             }
         });
     }
+
+    @Override
+    public void onClickPurchase(CustomerInfo customerInfo, Double subtotal) {
+        try{
+            boolean valid = customerInfo.validateOrder();
+            if(valid && subtotal!= null && subtotal>0) {
```
Comment (patrick-fs) [inline java/app/src/main/java/com/fullstorydev/shoppedemo/ui/checkout/CheckoutFragment.java]: Nit: whitespace around the operator

File: java/app/src/main/java/com/fullstorydev/shoppedemo/ui/checkout/CheckoutFragment.java [hunk 2]
```diff
@@ -60,8 +62,25 @@ public void onActivityCreated(@Nullable Bundle savedInstanceState) {
...
+                // placeholder for your logic here to complete purchase
+                Toast.makeText(getContext(), "Purchase success!", Toast.LENGTH_LONG).show();
+            }else{
```
Comment (patrick-fs) [inline java/app/src/main/java/com/fullstorydev/shoppedemo/ui/checkout/CheckoutFragment.java]: nit: whitespace around `else`

File: java/app/src/main/java/com/fullstorydev/shoppedemo/ui/checkout/CheckoutFragment.java [hunk 3]
```diff
@@ -60,8 +62,25 @@ public void onActivityCreated(@Nullable Bundle savedInstanceState) {
...
+                throw new IllegalArgumentException("Order not valid");
+            }
+        }catch (Exception e){
```
Comment (patrick-fs) [inline java/app/src/main/java/com/fullstorydev/shoppedemo/ui/checkout/CheckoutFragment.java]: nit: whitespace before `catch`

File: java/app/src/main/java/com/fullstorydev/shoppedemo/ui/checkout/CheckoutFragment.java (hunk 1 above)
Comment (sabrina-li) [reply java/app/src/main/java/com/fullstorydev/shoppedemo/ui/checkout/CheckoutFragment.java]: 🙏 gonna need to add lint for this project at some point..

**Pull request merged**